components/validator/l2_output_submitter.go

```diff
@@ -268,31 +284,59 @@ func (l *L2OutputSubmitter) fetchBlockNumbers() (*big.Int, *big.Int, error) {
 func (l *L2OutputSubmitter) waitL2Blocks(currentBlockNumber *big.Int, targetBlockNumber *big.Int) {
 	l.log.Info("validator submission interval has not elapsed", "currentBlockNumber", currentBlockNumber, "targetBlockNumber", targetBlockNumber)
 	waitBlockNum := new(big.Int).Sub(targetBlockNumber, currentBlockNumber)
-	waitSec := new(big.Int).Mul(waitBlockNum, l.l2BlockTime)
-	if waitSec.Cmp(common.Big0) == -1 {
-		waitSec = common.Big1
+
+	var waitDuration time.Duration
+	if waitBlockNum.Cmp(common.Big0) == -1 {
+		waitDuration = l.cfg.OutputSubmitterRetryInterval
+	} else {
+		waitDuration = time.Duration(new(big.Int).Mul(waitBlockNum, l.l2BlockTime).Uint64()) * time.Second
 	}
 
-	l.log.Info("wait for L2 blocks proceeding", "waitSec", waitSec)
-	l.retryAfter(time.Duration(waitSec.Uint64()) * time.Second)
+	l.log.Info("wait for L2 blocks proceeding", "waitDuration", waitDuration)
+	l.retryAfter(waitDuration)
+}
+
+type roundInfo struct {
+	isPublicRound       bool
+	isPriorityValidator bool
 }
 
-func (l *L2OutputSubmitter) isNextValidator() (bool, error) {
+// fetchCurrentRound fetches next validator address from ValidatorPool contract.
+// It returns if current round is public round, and if selected for priority validator if it's priority round.
+func (l *L2OutputSubmitter) fetchCurrentRound() (*roundInfo, error) {
```

Comment:
Since two booleans can fit within a 64-bit register, please simply return by copying.

```suggestion
func (l *L2OutputSubmitter) fetchCurrentRound() (roundInfo, error) {
```